### DISCUSSION TITLE
Unable to install firedrake - error installing loopy

### DISCUSSION THREAD

**Discussion by vcarlier** [firedrakeproject/firedrake#3090] [Firedrake support] [answered]
Hello everyone,

I am trying to install firedrake running the command `python3 firedrake-install` with python3.11.4

I have the following error message when installing loopy : 

```
Installing loopy/
Traceback (most recent call last):
  File "/home/vcarlier/firedrake-install", line 1917, in <module>
    install("loopy/")
  File "/home/vcarlier/firedrake-install", line 1058, in install
    run_pip_install(["-e", package])
  File "/home/vcarlier/firedrake-install", line 979, in run_pip_install
    check_call(pipinstall + pipargs)
  File "/home/vcarlier/firedrake-install", line 666, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/vcarlier/firedrake/bin/python', '-m', 'pip', 'install', '--no-build-isolation', '--no-binary', 'mpi4py,randomgen,islpy,numpy', '-vvv', '-e', 'loopy/']' returned non-zero exit status 1.


Install log saved in /home/vcarlier/firedrake-install.log
```

Here are the logs of the installation

[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/12499402/firedrake-install.log)

It seems that the problem is comming from the installation if islpy.

Thank you in advance for your help!

**Comment by dham** [comment]
Yes, your compiler is failing during the islpy build, which is odd. You appear to be on Ubuntu, which is a platform we test. It's possible that there is something about islpy and the C++ compiler you are using. Can you reproduce this separately from the Firedrake install? E.g. just make a clean venv and run:
```
python -m pip install --no-binary islpy -vvv islpy
```

**Reply by vcarlier** [reply]
It seems that I have a similar error when trying to install islpy separately

Here are the logs of the error when trying to install islpy
[islpy-install.log](https://github.com/firedrakeproject/firedrake/files/12503444/islpy-install.log)

**Comment by vcarlier** [comment, marked as answer]
Hello,
I've updated all my system (apt-get update / apt-get upgrade) and now the installation is working.
I don't know which package was causing the trouble, because updating gnu-gcc or python-dev would not solve the problem.
